controller/lib/hal/hal_stm32.cpp

```diff
@@ -240,20 +240,20 @@ void HalApi::digitalWrite(BinaryPin pin, VoltageLevel value) {
  * System timer
  *
  * I use one of the basic timers (timer 6) for general system timing.
- * I configure it to count every microsecond and generate an interrupt
+ * I configure it to count every 100ns and generate an interrupt
  * every millisecond
  *
  * The basic timers (like timer 6) are documented in chapter 29 of
  * the reference manual
  *****************************************************************/
-static void InitSysTimer() {
+void HalApi::InitSysTimer() {
   // Enable the clock to the timer
   EnableClock(TIMER6_BASE);
 
   // Just set the timer up to count every microsecond.
   TimerRegs *tmr = TIMER6_BASE;
-  tmr->reload = 999;
-  tmr->prescale = (CPU_FREQ_MHZ - 1);
+  tmr->reload = 9999;
```

Comment:
Number of clock ticks (100ns) / microsecond - 1.  I'll add a comment

Reply:
I meant the 999. Which then becomes 9999. To me it looks like it was possibly just "some arbitrary sort of big number" :) Why not 999999?

Reply:
It's only a 16-bit timer, so 99999 doesn't fit.
I'm actually planning to change this again to improve the resolution more when I fix issue #329, but didn't want to get into that in this PR.  The only reason I messed with it at all here is that I was going to use the timer value to time something short and needed more resolution, but in the end I didn't use it.

Reply:
Ok, I must not be very bright... still don't get what those 9s represent. I don't know what reload means and what units it is in. Does it mean the timer resets once it reaches so many ticks? Maybe then let's name it "reset_in_ticks"? Is it in ns? But then a us=1000ns so reset every us would mean 1000 and not 10000. But if the clock is 100ns, then a microsecond is 10 ticks, so the 10000 still doesn't make sense. But in your comment you are dividing by (ms-1) and not (us-1), which I also don't follow. So I will concluse that maybe the way it does it in the circuitry is weird and you are trying to translate your intent into how it does it nonintuitively on the hardware level. Perhaps this whole timer configuration routine deserves a function of its own that abstracts the confusing away and also unit tests it? Maybe this is a something worthy of a separate ticket rather than blocking this PR, though. How does that sound?